File: lint-arwaky/archive/contract/arch_compliance_port.py
"""arch_compliance_port — Port for architectural compliance coordinator."""

from abc import ABC, abstractmethod
from ..taxonomy import FilePath, ComplianceStatus, LintResultList

class IArchCompliancePort(ABC):
    """Port interface for orchestrating architectural compliance."""
    
    @abstractmethod
    async def scan(self, path: FilePath) -> LintResultList:
        """Perform architectural scan."""
        ...

    @abstractmethod
    async def apply_fix(self, path: FilePath) -> ComplianceStatus:
        """Apply architectural fixes."""
        ...

File: lint-arwaky/src-rust/contract/arch_compliance_port.rs
use async_trait::async_trait;
use super::*;

#[async_trait]
pub trait IArchCompliancePort: Send + Sync {
    async fn scan(&self, path: &FilePath) -> LintResultList;
    async fn apply_fix(&self, path: &FilePath) -> ComplianceStatus;
}
